mlflow/projects/utils.py

```diff
@@ -0,0 +1,133 @@
+import os
+import logging
+
+from mlflow.exceptions import ExecutionException
+
+_logger = logging.getLogger(__name__)
+
+
+MLPROJECT_PARAMETER_TYPES = ('string', 'float', 'path', 'uri')
+BAD_MLPROJECT_MESSAGE = "Invalid MLproject file: {}"
+MLPROJECT_MESSAGE = "MLproject file: %s"
+
+
+def validate_conda_env_path(path):
+    if not os.path.exists(path):
+        bad_conda_path_message = "conda environment file {} not found".format(path)
+        raise ExecutionException(
+            BAD_MLPROJECT_MESSAGE.format(bad_conda_path_message)
+        )
+
+
+def validate_project_yaml(project_yaml):
+    """validate the yaml in an MLproject file"""
+    project_allowed_entries = ('name', 'conda_env', 'docker_env', 'entry_points')
+
+    project_entries = project_yaml.keys()
+
+    # make sure there are no extraneous entries
+    _validate_entries_are_allowed('project-level', project_entries, project_allowed_entries)
+
+    if 'name' in project_entries:
+        _validate_name_yaml(project_yaml.get('name'))
+
+    if 'docker_env' in project_entries:
+        _validate_docker_env_yaml(project_yaml.get('docker_env'))
+        # disallow conda and docker envs together in the same MLproject file
+        if 'conda_env' in project_entries:
+            multi_env_message = "cannot specify conda_env and docker_env in the same project"
+            raise ExecutionException(BAD_MLPROJECT_MESSAGE.format(multi_env_message))
+
+    if 'conda_env' in project_entries:
+        _validate_conda_env_yaml(project_yaml.get('conda_env'))
+
+    if 'entry_points' in project_entries:
+        for name, entry_point in project_yaml.get('entry_points').items():
+            _validate_entry_point_yaml(name, entry_point)
+
+
+def _validate_entries_are_allowed(yaml_level, present_entries, allowed_entries):
+    """helper to validate that entry keys are in an allowable set"""
+    unparsed_entries = set(present_entries) - set(allowed_entries)
+    if unparsed_entries:
+        message = "{} entries {{{}}} will not be parsed by MLflow projects"\
```

Comment:
As described above, would be nice to include the list of known MLproject fields (docker_env, conda_env, etc) in this logged message